client/packages/system/src/Item/context/useUnitVariant.tsx

```diff
@@ -1,71 +1,79 @@
-import React, { PropsWithChildren, createContext, useContext } from 'react';
-import { useItemVariants } from '../api';
+import { useUnitVariantList } from '../api';
+import { create } from 'zustand';
+import isEqual from 'lodash/isEqual';
+import { UnitVariantNode } from '@common/types';
 import { NumUtils } from '@common/utils';
+import { useEffect } from 'react';
 
-export const useUnitVariant = (itemId?: string) => {
-  const { data } = useItemVariants();
-  const item = data?.find(item => item.id === itemId);
-
-  const asPackUnit = (packSize: number) => {
-    return (
-      item?.variants.find(variant => variant.packSize === packSize)
-        ?.shortName || `${item?.unit} x ${packSize}`
-    );
-  };
-
-  // calculates SOH, AMC, MOS, Target, Suggested and Requested based on default or user selected variant
-  const numberOfPacksFromQuantity = (
-    totalQuantity: number,
-    packSize: number
-  ) => {
-    return NumUtils.round(totalQuantity / packSize, 2);
-  };
-
-  const options = item?.variants ?? [];
-
-  // const quantityFromNumberOfPacks - reverse of above
-
-  // set in app data
-  const setDefaultOption = () => {};
-
-  const defaultOption = () => {
-    // TODO: search app data first to see if user has preferred preference
-    return item?.variants.find(variant =>
-      variant.longName.includes(item?.unit)
-    );
+interface UnitState {
+  // From back end
+  items: {
+    [itemId: string]: UnitVariantNode;
   };
+  // Should be called on startup when fetching multi unit variants
+  setItems: (newItems: UnitVariantNode[]) => void;
+}
 
+const useUnitStore = create<UnitState>(set => {
   return {
-    asPackUnit,
-    numberOfPacksFromQuantity,
-    options,
-    setDefaultOption,
-    defaultOption,
+    items: {},
+    userSelected: {},
+    // TODO add user selected
+
+    setItems: newItems =>
+      set(() => {
+        const items = newItems.reduce(
+          (acc, item) => ({ [item.itemId]: item, ...acc }),
+          {}
+        );
+        return { items };
+      }),
   };
+});
+
+export const useInitUnitStore = () => {
+  const { setItems } = useUnitStore();
+  // This should happen on startup and when store is changed (for store changed, calculated mostUsed and app data userSelecte would change)
```

Comment:
I renamed file so can't apply your suggestion: a585a3af0e34cade119c4b29c3410c7fd79982b3

Reply:
> not sure yet what 'app data userSelected' is referring to? hopefully it will become clear as I read on

It's not used yet, should be clear in a PR or two (you've commented about it in the epic, i had it as 'default' but it was more of a user selected. (I'll mention this question in the PR this is implemented in)